.github/workflows/gh-env-cleanup.yml

```diff
@@ -0,0 +1,108 @@
+name: GitHub Environment Cleanup
+
+on:
+  schedule:
+    - cron: '0 0 * * 1-5' # Weekdays at midnight UTC
+  pull_request:
+    types: [closed] # Trigger on PR close
+
+jobs:
+  # Scheduled cleanup job
+  scheduled-cleanup:
+    if: github.event_name == 'schedule'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.aws_access_key_id }}
+          aws-secret-access-key: ${{ secrets.aws_secret_access_key }}
+          aws-region: "us-gov-west-1"
+
+      - name: Get Bot Token from Parameter Store
+        uses: marvinpinto/action-inject-ssm-secrets@latest
+        with:
+          ssm_parameter: /devops/VA_VSP_BOT_GITHUB_TOKEN
+          env_variable_name: VA_VSP_BOT_GITHUB_TOKEN
+
+      - name: Set Up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.4'
+
+      - name: Clean Up Environments
+        run: |
+          cat << 'EOF' | ruby
+            require 'bundler/inline'
+            gemfile { source 'https://rubygems.org'; gem 'octokit' }
+
+            CURRENT_TIME = Time.now.to_i
+            THRESHOLD_SECONDS = 90 * 24 * 60 * 60 # 90 days in seconds
+            REPO = ENV['GITHUB_REPOSITORY']
+            CLIENT = Octokit::Client.new access_token: ENV['VA_VSP_BOT_GITHUB_TOKEN']
+
+            def old_and_unprotected? env
+              env[:protection_rules].empty? && CURRENT_TIME - env[:created_at].to_i > THRESHOLD_SECONDS
+            end
+
+            def delete_environment env
+              puts "\nDeleting: #{env[:name]}"
+              CLIENT.delete_environment REPO, URI.encode_www_form_component(env[:name])
+            end
+
+            print 'Fetching and deleting old environments'
+            100.times do |page|
```

Comment:
Typically we'll only have 1 page worth of environments open so I probably don't need to loop but if this ever needs to run on thousands of envs again, this should run through them in 1-2 runs (rate limiting aside).